pkg/scheduler/algorithm/priorities/interpod_affinity.go

```diff
@@ -94,33 +83,100 @@ func (p *podAffinityPriorityMap) processTerms(terms []v1.WeightedPodAffinityTerm
 	return nil
 }
 
-// CalculateInterPodAffinityPriority compute a sum by iterating through the elements of weightedPodAffinityTerm and adding
-// "weight" to the sum if the corresponding PodAffinityTerm is satisfied for
-// that node; the node(s) with the highest sum are the most preferred.
-// Symmetry need to be considered for preferredDuringSchedulingIgnoredDuringExecution from podAffinity & podAntiAffinity,
-// symmetry need to be considered for hard requirements from podAffinity
-func (ipa *InterPodAffinity) CalculateInterPodAffinityPriority(pod *v1.Pod, sharedLister schedulerlisters.SharedLister, nodes []*v1.Node) (framework.NodeScoreList, error) {
+// CalculateInterPodAffinityPriorityMap calculate the number of matching pods on the passed-in "node",
+// and return the number as Score.
+func CalculateInterPodAffinityPriorityMap(pod *v1.Pod, meta interface{}, nodeInfo *schedulernodeinfo.NodeInfo) (framework.NodeScore, error) {
+	node := nodeInfo.Node()
+	if node == nil {
+		return framework.NodeScore{}, fmt.Errorf("node not found")
+	}
+
+	var topologyScore topologyPairToScore
+	if priorityMeta, ok := meta.(*priorityMetadata); ok {
+		topologyScore = priorityMeta.topologyScore
+	}
+	if topologyScore == nil {
+		return framework.NodeScore{Name: node.Name, Score: 0}, nil
+	}
+
+	var score int64
+	for tpKey, tpValues := range topologyScore {
+		if v, exist := node.Labels[tpKey]; exist {
+			score += tpValues[v]
+		}
+	}
+
+	return framework.NodeScore{Name: node.Name, Score: score}, nil
+}
+
+// CalculateInterPodAffinityPriorityReduce normalizes the score for each filteredNode,
+// The basic rule is: the bigger the score(matching number of pods) is, the smaller the
+// final normalized score will be.
+func CalculateInterPodAffinityPriorityReduce(pod *v1.Pod, meta interface{}, sharedLister schedulerlisters.SharedLister,
+	result framework.NodeScoreList) error {
+	var topologyScore topologyPairToScore
+	if priorityMeta, ok := meta.(*priorityMetadata); ok {
+		topologyScore = priorityMeta.topologyScore
+	}
+	if topologyScore == nil {
```

Comment:
use ```len(topologyScore) == 0``` instead because in some cases we may allocate the map, but it will be empty, and so the reduce logic below will not necessary.